### ISSUE TITLE
`make test` fails due to Swift Language Version = 3.0 in TestFramework3

### ISSUE THREAD

**Issue by sstadelman** [Carthage/Carthage#2780]
**EDIT:  I see now in the `fixtures/DependencyTest.zip` archive.  Will validate & submit PR.**

* carthage install method: [ ] .pkg, [ ] homebrew, [x] source (dae36671bf444aad0338878523a54c8c255b42e1)
* `which carthage`: n/a
* `carthage version`:  n/a
* `xcodebuild -version`: Xcode 10.2.1, Build version 10E1001
* Are you using `--no-build`? n/a
* Are you using `--no-use-binaries`? n/a
* Are you using `--use-submodules`? n/a
* Are you using `--cache-builds`?  n/a
* Are you using `--new-resolver`? n/a

I'm working on validating the netrc parser implementation proposed in #2774, as this is a dependency for SAP.  I'm seeing a test failure in `CarthageKitTests.ProjectSpec.buildCheckedOutDependencies`, where the TestFramework3 fails compilation due to the **Swift Language Version: 3.0**.

```
Test Suite 'ProjectSpec' started at 2019-05-01 11:41:41.070
Test Case '-[CarthageKitTests.ProjectSpec buildCheckedOutDependenciesWithOptions, should determine build order without repo cache]' started.
2019-05-01 11:41:43.768 xctest[40250:6160271] Building scheme "TestFramework3_Mac" in TestFramework3.xcodeproj
~/github2/Carthage/Tests/CarthageKitTests/ProjectSpec.swift:32: error: -[CarthageKitTests.ProjectSpec buildCheckedOutDependenciesWithOptions, should determine build order without repo cache] : failed - expected to be nil, got <Build FailedTask failed with exit code 65:/usr/bin/xcrun xcodebuild -project ~/github2/Carthage/.build/x86_64-apple-macosx/debug/CarthagePackageTests.xctest/Contents/Resources/DependencyTest/Carthage/Checkouts/TestFramework3/TestFramework3.xcodeproj -scheme TestFramework3_Mac -configuration Debug -derivedDataPath ~/Library/Caches/org.carthage.CarthageKit/DerivedData/10.2.1_10E1001/TestFramework3/v1.0 ONLY_ACTIVE_ARCH=NO CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES archive -archivePath /var/folders/4x/kdqw8pjs3hd0rrq8cbh5tv5r0000gn/T/TestFramework3 SKIP_INSTALL=YES GCC_INSTRUMENT_PROGRAM_FLOW_ARCS=NO CLANG_ENABLE_CODE_COVERAGE=NO STRIP_INSTALLED_PRODUCT=NO (launched in ~/github2/Carthage/.build/x86_64-apple-macosx/debug/CarthagePackageTests.xctest/Contents/Resources/DependencyTest/Carthage/Checkouts/TestFramework3)This usually indicates that project itself failed to compile.>
```

**Comment by sstadelman** [comment]
This fix is straightforward:  I updated the `TestFrameworkX` targets' `Swift LanguageVersion` to `Swift 4.2` (Not sure what backwards-compatibility should be enforced, this seemed reasonable?).

I'm now seeing a very unexpected `error: filename "File.swift" used twice...` message.

```
<unknown>:0: error: filename "File.swift" used twice: '~/github2/CarthageFork/Tests/CarthageKitTests/fixtures/DependencyTest/SourceRepos/TestFramework1/File.swift' and 
'~/github2/CarthageFork/Tests/CarthageKitTests/fixtures/DependencyTest/SourceRepos/TestFramework2/File.swift'
<unknown>:0: note: filenames are used to distinguish private declarations with the same name
```

There is a comment at the head of each `File.swift`: `// Make sure this is compiled as a Swift framework.`, leading me to believe this is not an unknown state.  Any thoughts?

**Comment by sstadelman** [comment]
Resolved by #2781
